akvo/rsr/forms.py

```diff
@@ -110,56 +75,17 @@ class RegisterForm(PasswordValidationMixin, forms.Form):
             attrs={'placeholder': _('Last name')}
         ),
     )
-    password1 = forms.CharField(
-        label=_('Password'),
-        widget=forms.PasswordInput(
-            attrs={'placeholder': _('Password')},
-            render_value=False
-        )
-    )
-    password2 = forms.CharField(
-        label=_('Repeat password'),
-        widget=forms.PasswordInput(
-            attrs={'placeholder': _('Repeat password')},
-            render_value=False
-        )
-    )
 
-    def clean_email(self):
-        """
-        Verify that the email entered does not exist as an email or username.
-        """
-        email = self.cleaned_data['email']
-        if get_user_model().objects.filter(email__iexact=email).exists() \
-                or get_user_model().objects.filter(username__iexact=email).exists():
-            raise forms.ValidationError(_('A user with this email address already exists.'))
-        return email
-
-    def clean_password2(self):
-        password1 = self.cleaned_data.get('password1')
-        password2 = self.cleaned_data.get('password2')
-        if password1 and password2:
-            if password1 != password2:
-                raise forms.ValidationError(
-                    _('Passwords do not match. Please enter the same password in both fields.')
-                )
-        return password2
+    class Meta:
+        model = User
+        fields = ('email', 'first_name', 'last_name', 'password1', 'password2')
 
     def save(self, request):
-        """Create the new user and RegistrationProfile, and returns the user."""
-
-        User = get_user_model()
-        username = email = self.cleaned_data['email']
-        password = self.cleaned_data['password1']
-        first_name = self.cleaned_data['first_name']
-        last_name = self.cleaned_data['last_name']
-        user = User.objects.create_user(
-            username, email, password, first_name=first_name, last_name=last_name
-        )
-
+        self.instance.username = self.cleaned_data['email']
+        self.instance.last_login = self.instance.date_joined
```

Comment:
No, registration does not log users into the system. I think setting the `last_login` attribute is a trick so that newly registered user can reset their password.

The old code calls the `User.objects.create_user` method which internally calls the `CustomUserManager._create_user` which [set the `last_login` attribute upon user creation](https://github.com/akvo/akvo-rsr/blob/4955-password-policy/akvo/rsr/models/user.py#L50). The `last_login` attribute is then [used as a filter](https://github.com/akvo/akvo-rsr/blob/4955-password-policy/akvo/rsr/forms.py#L111-L113) in the `PasswordResetForm` so the newly registered users are included. There is also [a test case](https://github.com/akvo/akvo-rsr/blob/4955-password-policy/akvo/rsr/tests/views/test_account.py#L275-L289) to make sure the behavior works.